Button and alert

manujagobind opened the pull request:
**Summary**

This change adds the following:
- Set up initial directory structure for the component library
- Initial set of styling for core design aspects to be used across multiple components
- Styling for buttons component
- Documentation for buttons component
- Styling for alerts component
- Documentation for alerts component
- Index page to capture the list of all components

**Third Party Software Used**

This change makes use of the following external code:
- Uses [FontAwesome v5.15](https://fontawesome.com/v5.15/icons) for icons
- Uses [HighlightJS](https://highlightjs.org/) for code highlighting in the documentation pages

**View Live**
Please refer to the Netlify section below for the preview link.

manujagobind committed:
Add buttons

manujagobind committed:
Improve file structure, add auth & desc meta tags to HTML

manujagobind committed:
Add support for icons

manujagobind committed:
Add usage

manujagobind committed:
Add demo link to README

manujagobind committed:
Fix typo in GitHub profile URL

manujagobind committed:
[fix] allow for scroll-x on mobile for demo container

manujagobind committed:
[feat] Add alerts component

manujagobind committed:
[fix] Replace white with --ink-white in CSS

shrushti2000 reviewed (commented):
Looks good to me. You can add alerts with close buttons and floating action buttons too.

manujagobind commented:
> Looks good to me. You can add alerts with close buttons and floating action buttons too.

Hey, thanks @shrushti2000 for the review. For now I'm only restricting to HTML and CSS. Didn't add close option as it requires JS.

HarshnaNHaswani commented on src/button/button.html:
```diff
@@ -0,0 +1,52 @@
+                <a href="#" class="btn btn-outline">Outline</a>
+                <a href="#" class="btn btn-link">Link</a>
+            </div>
+        </section>
```
If you want to link the button to something, nest the `anchor` inside the `button` and give the necessary classes to the button. Otherwise, it is not a button. It is just an `<a>` that looks like a `<button>` and will not have the default properties of a button

HarshnaNHaswani commented on src/pages/button.html:
```diff
@@ -25,30 +25,83 @@ <h2>Basic</h2>
+                Danger
+            </button>
+        </div>
+        </section>
```
This can be treated as a second example/ sub-category for icon buttons. Just tell your user what changes you made in the icon tag to get this result (change: icon, color/background-color/outline color)

HarshnaNHaswani commented on src/styles/variables.css:
```diff
@@ -3,6 +3,8 @@
     
     --ink-red: #d43324;
     --ink-green: #14c2ad;
```
Use a standard green for success. Familiarity helps. This turquoise bluish color is quite sharp and the text clarity is a bit low. You can use the dev tool's inspector to check whether the contrast between the text and the background is agreeable. Choose your colors wisely

manujagobind replied:
I haven't worked on the typography yet. Most of the colors I've selected are from the ones used in [this site](https://flutter.dev). Maybe when I work on typography things would start looking better. 

Regarding green color, I'll check. Thanks.

HarshnaNHaswani replied:
I'm talking about the clash between the text color and background color.

HarshnaNHaswani replied:
Thanks for looking into the suggestions and responding!

HarshnaNHaswani commented on src/pages/alert.html:
```diff
@@ -0,0 +1,88 @@
+            <div class="container code">
+                <pre>
+                    <code class="language-html">
+    &lt;div class="alert alert-info"&gt;
+        &lt;i class="fas fa-question-circle"&gt;&lt;/i&gt;
+        &lt;p&gt;This is an info alert.&lt;/p&gt;
+    &lt;/div&gt;
+    &lt;div class="alert alert-warn"&gt;
+        &lt;i class="fas fa-exclamation-circle"&gt;&lt;/i&gt;
+        &lt;p&gt;This is a warning alert.&lt;/p&gt;
+    &lt;/div&gt;
+    &lt;div class="alert alert-success"&gt;
+        &lt;i class="fas fa-check-circle"&gt;&lt;/i&gt;
+        &lt;p&gt;This is a success alert.&lt;/p&gt;
+    &lt;/div&gt;
+    &lt;div class="alert alert-danger"&gt;
+        &lt;i class="fas fa-times-circle"&gt;&lt;/i&gt;
+        &lt;p&gt;This is a danger alert.&lt;/p&gt;
+    &lt;/div&gt;
```
format code here

manujagobind replied:
Hey @HarshnaNHaswani - if you're referring to this part of code being on the left of the page - then it's intentional. The `pre` tags remembers the spaces. If I move this section to the right then that extra space will show up in my code container as well in the docs.

HarshnaNHaswani replied:
My mistake, I forgot. Sorry!

HarshnaNHaswani reviewed (changes requested):
It's awesome that you have followed semantic tags almost everywhere. Please give the suggestions I mentioned a thought before proceeding

manujagobind committed:
Add alerts with close option

manujagobind committed:
Use a familiar green color

manujagobind committed:
Refactor button documentation based on inputs from review comments

manujagobind commented:
Hey @HarshnaNHaswani, @shrushti2000 - have addressed the comments.

@HarshnaNHaswani - regarding the contrast b/w text and background, I'll check that once I work on typography.

manujagobind committed:
Update alerts doc with JS imports
